### PR TITLE
backport: bitcoin#22630, 22577, 22277

### PR DESCRIPTION
**Pull request by vijaydasmp**
bitcoin backports

### CODE REVIEW

**Comment by vijaydasmp** [comment]
Hello @UdjinM6 for review

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

File: src/test/miner_tests.cpp [hunk 1]
```diff
@@ -23,6 +23,7 @@
 #include <util/system.h>
 #include <util/time.h>
 #include <validation.h>
+#include <versionbits.h>
```
[the remark this answers is not in the document]
Comment (vijaydasmp) [reply src/test/miner_tests.cpp:26]: tried to bring 22277

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
pls see d07dee1ebf

File: src/test/miner_tests.cpp [hunk 2]
```diff
@@ -240,24 +229,21 @@ BOOST_AUTO_TEST_CASE(CreateNewBlock_validity)
...
             LOCK(cs_main);
-            pblock->nVersion = 2;
+            pblock->nVersion = VERSIONBITS_TOP_BITS;
             pblock->nTime = m_node.chainman->ActiveChain().Tip()->GetMedianTimePast()+1;
             CMutableTransaction txCoinbase(*pblock->vtx[0]);
             txCoinbase.nVersion = 1;
-            txCoinbase.vin[0].scriptSig = CScript() << (m_node.chainman->ActiveChain().Height() + 1);
-            txCoinbase.vin[0].scriptSig.push_back(blockinfo[i].extranonce);
-            txCoinbase.vin[0].scriptSig.push_back(m_node.chainman->ActiveChain().Height());
-            txCoinbase.vout[0].scriptPubKey = CScript();
```
Comment (UdjinM6) [inline src/test/miner_tests.cpp:250]: should keep `txCoinbase.vout[0].scriptPubKey = CScript();`

File: src/test/miner_tests.cpp [hunk 3]
```diff
@@ -240,24 +229,21 @@ BOOST_AUTO_TEST_CASE(CreateNewBlock_validity)
...
+            txCoinbase.vin[0].scriptSig = CScript() << (m_node.chainman->ActiveChain().Height() + 1) << blockinfo[i].extranonce;
```
Comment (UdjinM6) [inline src/test/miner_tests.cpp:236]:
```suggestion
            txCoinbase.vin[0].scriptSig = CScript() << (m_node.chainman->ActiveChain().Height() + 1) << BLOCKINFO[i].extranonce;
```

**Review by UdjinM6 — CHANGES_REQUESTED** [review]
You missed a couple of things it seems

(btw, no need to copy/paste or reimplement patches, you can simply `fetch` + `cherry-pick` them)

**Comment by vijaydasmp** [comment]
> You missed a couple of things it seems
> 
> (btw, no need to copy/paste or reimplement patches, you can simply `fetch` + `cherry-pick` them)

Thanks @UdjinM6 , will do it from now

**Review by UdjinM6 — CHANGES_REQUESTED** [review]

**Review by UdjinM6 — APPROVED** [review]
utACK 123b034b4997745a3920cdc5efcd507796eebd86

**Pull request merged**